api/routes.py

```
@@ -0,0 +1,294 @@
+async def analyze_food_by_text(
+    request: FoodAnalysisRequest, gemini: GeminiService = Depends(get_gemini_service)
+):
+    """Analyze food from text description."""
+    logger.info(f"Analyzing food from text: {request.description[:50]}...")
+    try:
+        result = await gemini.analyze_food_by_text(request.description)
+        logger.info(f"Successfully analyzed food: {result.food_name}")
+        return result
+    except GeminiServiceException as e:
+        logger.error(f"Gemini service error while analyzing food text: {str(e)}")
+        raise HTTPException(status_code=e.status_code, detail=e.message)  # pragma: no cover
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)